app/src/main/java/pl/droidevs/books/login/LoginActivity.java

```diff
@@ -1,37 +1,87 @@
 package pl.droidevs.books.login;
 
+import android.arch.lifecycle.ViewModelProvider;
+import android.arch.lifecycle.ViewModelProviders;
 import android.content.Intent;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
-import android.widget.TextView;
+import android.widget.Button;
+import android.widget.EditText;
+
+import javax.inject.Inject;
 
 import butterknife.BindView;
 import butterknife.ButterKnife;
 import butterknife.OnClick;
+import butterknife.OnTextChanged;
 import butterknife.Unbinder;
 import dagger.android.AndroidInjection;
 import pl.droidevs.books.R;
 import pl.droidevs.books.library.LibraryActivity;
 
 public class LoginActivity extends AppCompatActivity {
 
-    @BindView(R.id.textHello)
-    TextView textHello;
+    @BindView(R.id.login_edit_text)
+    EditText loginEditText;
+
+    @BindView(R.id.login_button)
+    Button loginButton;
 
     private Unbinder unbinder;
 
+    @Inject
+    ViewModelProvider.Factory viewModelFactory;
+
+    private LoginViewModel loginViewModel;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
 
         AndroidInjection.inject(this);
         this.unbinder = ButterKnife.bind(this);
+
+        setupViewModel();
+
+        if (this.loginViewModel.isLoggedIn()) {
+            showLibraryActivity();
+        }
+
+        manageLoginButton();
+    }
+
+    private void setupViewModel() {
+        this.loginViewModel = ViewModelProviders
+                .of(this, this.viewModelFactory)
+                .get(LoginViewModel.class);
+    }
+
+    private void showLibraryActivity() {
+        startActivity(new Intent(this, LibraryActivity.class));
+        finish();
+    }
+
+    private void manageLoginButton() {
+
+        if (this.loginViewModel.isInputValid(this.loginEditText.getText().toString())) {
+            this.loginButton.setEnabled(true);
+
+            return;
```

Comment:
A co powiesz na zmienną isValid i przypisanie jej do loginButton.setEnabled? Pozbędziemy się wtedy istrukcji warunkowej.

Reply:
Omg.. Indeed. Jak to czasami najbardziej oczywiste rzeczy mózg potrafi skomplikować... muszę bardziej na to zwracać uwagę *Shame on me*